### PR TITLE
Drop support for older rocm releases - 4.5.0 till 5.0.2

### PR DESCRIPTION
**Pull request by srekolam**
this PR is meant to drop support for the older rocm-releases from 4.5.0 till 5.0.2

### CODE REVIEW

**Comment by srekolam** [comment]
@spackbot re-run pipeline:

File: var/spack/repos/builtin/packages/comgr/package.py [hunk 1]
```diff
@@ -27,10 +27,25 @@ class Comgr(CMakePackage):
...
     version("5.1.3", sha256="3078c10e9a852fe8357712a263ad775b15944e083f93a879935c877511066ac9")
     version("5.1.0", sha256="1cdcfe5acb768ef50fb0026d4ee7ba01e615251ad3c27bb2593cdcf8c070a894")
-    version("5.0.2", sha256="20d733f70d8edb573d8c92707f663d7d46dcaff08026cd6addbb83266679f92a")
-    version("5.0.0", sha256="da1bbc694bd930a504406eb0a0018c2e317d8b2c136fb2cab8de426870efe9a8")
-    version("4.5.2", sha256="e45f387fb6635fc1713714d09364204cd28fea97655b313c857beb1f8524e593")
-    version("4.5.0", sha256="03c5880e0922fcff31306f7da2eb9d3a3709d9b5b75b3524dcfae85f4b181678")
+    version(
+        "5.0.2",
+        sha256="20d733f70d8edb573d8c92707f663d7d46dcaff08026cd6addbb83266679f92a",
+    )
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/comgr/package.py]: You  forgot the `deprecated` option here.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Looks like one of the deprecated options is missing.

**Comment by cgmb** [comment]
What's the criteria for deciding when and what to deprecate?

**Comment by srekolam** [comment]
> What's the criteria for deciding when and what to deprecate?

@cgmb , usually i email and discuss in packaging meeting to check if older releases can be deprecated.

**Comment by srekolam** [comment]
@tldahlgren can i get this reviewed and merged if things look okay.

**Review by tldahlgren — APPROVED** [review]
LGTM

**Pull request merged**